x-pack/platform/plugins/shared/lens/public/datasources/form_based/esql_layer/text_based_languages.tsx

```diff
@@ -660,7 +659,18 @@ export function getTextBasedDatasource({
       references1: SavedObjectReference[],
       persistableState2: TextBasedPersistedState,
       references2: SavedObjectReference[]
-    ) => isEqual(persistableState1, persistableState2),
+    ) =>
+      // undefined is not equal to missing
+      isEqual(
+        {
+          initialContext: undefined,
+          ...persistableState1,
+        },
+        {
+          initialContext: undefined,
+          ...persistableState2,
+        }
+      ),
```

Comment:
This fixes and issue when you edit a non-new ES|QL panel, as soon as thee flyout is opened the state is dirty with no change. The cause is that `lodash.isEqual` is called on 2 objects on with `initialContext` property missing, and the other with it set to `undefined`. This causes the return to be `false` when it should be `true`.